src/wallet/wallet.cpp

```diff
@@ -5828,32 +5822,22 @@ bool CWallet::CommitTransaction(CWalletTx& wtxNew, std::optional<std::reference_
     return true;
 }
 
-CAmount CWallet::GetRequiredFee(unsigned int nTxBytes)
+CAmount CWallet::ConstrainFee(CAmount requestedFee, unsigned int nTxBytes)
 {
-    return std::max(minTxFee.GetFee(nTxBytes), ::minRelayTxFee.GetFeeForRelay(nTxBytes));
+    return std::min(std::max(::minRelayTxFee.GetFeeForRelay(nTxBytes), requestedFee), maxTxFee);
 }
 
-CAmount CWallet::GetMinimumFee(unsigned int nTxBytes, unsigned int nConfirmTarget, const CTxMemPool& pool)
+CAmount CWallet::GetMinimumFee(const CTransaction& tx, unsigned int nTxBytes)
 {
     // payTxFee is user-set "I want to pay this much"
     CAmount nFeeNeeded = payTxFee.GetFee(nTxBytes);
     // user selected total at least (default=true)
     if (fPayAtLeastCustomFee && nFeeNeeded > 0 && nFeeNeeded < payTxFee.GetFeePerK())
         nFeeNeeded = payTxFee.GetFeePerK();
```

Comment:
```suggestion
    CAmount nFeeNeeded = payTxFee.GetFee(std::min(1000, nTxBytes));
```
Proof that this is equivalent:
* `fPayAtLeastCustomFee` is initialized to `true` and nothing ever changes it.
* `payTxFee` is nonnegative and `nTxBytes` is positive. Therefore `nFeeNeeded` is nonnegative, and can only be zero if `payTxFee` is zero.
* if `payTxFee` is zero then `payTxFee.GetFee(std::min(1000, nTxBytes)) == payTxFee.GetFee(nTxBytes)`.
* otherwise, the `if` statement is equivalent to
  ```c++
  if (nFeeNeeded < payTxFee.GetFee(1000))
      nFeeNeeded = payTxFee.GetFee(1000);
  ```
  and since `GetFee` is an increasing function, this is equivalent to the suggestion.

Reply:
Shouldn’t this use `max` instead of `min`?

Reply:
Also, is the 1000 simply 1 KiB, or is there a deeper significance that would justify naming that constant?

Also, what about `CAmount nFeeNeeded = std::max(payTxFee.GetFeePerK(), payTxFee.GetFee(nTxBytes));`, which might make the 1 KiB thing clearer. Or
```c++
constexpr uint8_t B = 1;
constexpr uint8_t KiB = 1000 * B;
...
CAmount nFeeNeeded = payTxFee.GetFee(std::min(1*KiB, nTxBytes));
```

Reply:
> Shouldn’t this use max instead of min?

Yes, that was an error that I fixed concurrently with your comment.

> Also, is the 1000 simply 1 KiB, or is there a deeper significance that would justify naming that constant?

1000 is the fee rate denominator. It doesn't have a named constant anywhere else it is used.

Reply:
Oh right, I reversed my units 🤦🏼